src/main/java/com/influxdb/v3/client/internal/InfluxDBClientImpl.java

```diff
@@ -183,16 +185,52 @@ public Stream<Object[]> query(@Nonnull final String query,
                     List<FieldVector> fieldVectors = vector.getFieldVectors();
-                    return IntStream
-                            .range(0, vector.getRowCount())
-                            .mapToObj(rowNumber -> {
-
-                                ArrayList<Object> row = new ArrayList<>();
-                                for (FieldVector fieldVector : fieldVectors) {
-                                    row.add(fieldVector.getObject(rowNumber));
-                                }
-                                return row.toArray();
-                            });
+                    return IntStream.range(0, vector.getRowCount())
+                           .mapToObj(rowNumber -> {
+                               ArrayList<Object> row = new ArrayList<>();
+                               for (int i = 0; i < fieldVectors.size(); i++) {
+                                   var schema = vector.getSchema().getFields().get(i);
+                                   var metaType = schema.getMetadata().get("iox::column::type");
+                                   String valueType = metaType != null ? metaType.split("::")[2] : null;
+
+                                   if ("field".equals(valueType)) {
+                                       switch (metaType) {
+                                           case "iox::column_type::field::integer":
+                                           case "iox::column_type::field::uinteger":
+                                               var intValue = (Long) fieldVectors.get(i)
+                                                                                 .getObject(rowNumber);
+                                               row.add(intValue);
+                                               break;
+                                           case "iox::column_type::field::float":
+                                               var doubleValue = (Double) fieldVectors.get(i)
+                                                                                      .getObject(rowNumber);
+                                               row.add(doubleValue);
+                                               break;
+                                           case "iox::column_type::field::string":
+                                               var textValue = (Text) fieldVectors.get(i)
+                                                                                  .getObject(rowNumber);
+                                               row.add(textValue.toString());
+                                               break;
+                                           case "iox::column_type::field::boolean":
+                                               var boolValue = (Boolean) fieldVectors.get(i)
+                                                                                     .getObject(rowNumber);
+                                               row.add(boolValue);
```

Comment:
`fieldVectors.get(i).getObject(rowNumber)` is called on multiple place, move it before `switch`